python/GafferTest/FileSequencePathFilterTest.py

```diff
@@ -293,15 +293,30 @@ def setUp( self ) :
 
 		GafferTest.TestCase.setUp( self )
 
+		self.__dir = str( Gaffer.FileSystemPath( os.path.join( tempfile.gettempdir(), "gafferFileSequencePathFilterTest" ) ) )
+
 		# clear out old files and make empty directory
 		# to work in
 		if os.path.exists( self.__dir ) :
 			shutil.rmtree( self.__dir )
 		os.mkdir( self.__dir )
 
-		os.mkdir( self.__dir + "/dir" )
+		# On Windows, Python returns a temporary directory in Windows 8.3 file
+		# naming format, which is a shortened version of the full filename.
+		# This name contains numbers, which causes the test to fail when
+		# IECore::findSequences thinks it finds a sequence due to the number
+		# in the directory name.
```

Comment:
I ran into the same problem today, but on Linux. I believe it is a bug in FileSequencePathFilter, so we should fix that rather than work around it here. See https://github.com/GafferHQ/gaffer/pull/4871/commits/1bcfa5b5fcb0cbb98c5506eb6750f4e53969ae5e#diff-f039e155263dad7dfce3f08d329b1700c0ac36d831a398bddfb1f89d11ad8caeL45.

Reply:
I'll have a look at that and see what needs to change in Cortex.

Reply:
It looks like there's two possible places to fix this - I'm unsure of the expected behavior.

In Cortex, `IECore.findSequences( [ "/dir001/a.exr" ], 1 )` happily returns a sequence `/dir###/a.exr`.

In Gaffer, we look at the full path to see if it's a sequence :
https://github.com/GafferHQ/gaffer/blob/0d476a3d6ae20395df9d541baff4acfbe3270b09/src/Gaffer/FileSequencePathFilter.cpp#L112-L115

Should Cortex continue returning sequences if the number is in the directory, and we only give it the file name portion? Or is Cortex not meant to include sequences in directories, we fix it in Cortex and continue giving it full paths?

Reply:
I don't know whether the Cortex behaviour is intended or not to be honest. Skimming the unit tests I didn't see anything that was actively testing it. My guess would be that it was unintended, and I don't think it's what we want in Gaffer. But I suspect Image Engine in particular have a lot of infrastructure relying on it, so I'm not feelign quite gung-ho enough to suggest we change things. @ivanimanishi, perhaps you might know if you expect numbered directories to be treated as a sequence?

At the moment, I'm inclined to make the more isolated fix in Gaffer, and to add a todo suggesting that it be moved to Cortex at some point.

Reply:
Neither myself nor Petru are aware of any current use at IE where we rely on it finding a sequence of directories. In fact, we recently got to deal with things breaking because of that exact behaviour.

I wouldn't be against doing the fix in cortex, and get rid of it for good.